packages/ovault-evm/contracts/VaultComposerSyncHydra.sol

```diff
@@ -0,0 +1,214 @@
+     * @dev If the destination eid is different, it sends a LayerZero cross-chain transaction
+     * @param _oft The OFT contract address to use for sending
+     * @param _sendParam The parameters for the send operation
+     * @param _refundAddress Address to receive excess payment of the LZ fees
+     */
+    function _send(address _oft, SendParam memory _sendParam, address _refundAddress) internal override {
+        if (_sendParam.dstEid == VAULT_EID) {
+            /// @dev Can do this because _oft is validated before this function is called
+            address erc20 = _oft == ASSET_OFT ? ASSET_ERC20 : SHARE_ERC20;
+
+            if (msg.value > 0) revert NoMsgValueExpected();
+            IERC20(erc20).safeTransfer(_sendParam.to.bytes32ToAddress(), _sendParam.amountLD);
+        } else {
+            _sendOrFallbackToHub(_oft, _sendParam, _refundAddress);
+        }
+    }
+
+    /**
+     * @dev Internal function to refund input tokens to sender on source during a failed transaction
+     * @param _oft The OFT contract address used for refunding
+     * @param _message The original message that was sent
+     * @param _amount The amount of tokens to refund
+     * @param _refundAddress Address to receive the refund
+     */
+    function _refund(
+        address _oft,
+        bytes calldata _message,
+        uint256 _amount,
+        address _refundAddress
+    ) internal virtual override {
+        /// @dev Extract refund details from _message header (always present from lzReceive)
+        SendParam memory refundSendParam;
+        refundSendParam.dstEid = OFTComposeMsgCodec.srcEid(_message);
+        refundSendParam.to = OFTComposeMsgCodec.composeFrom(_message);
+        refundSendParam.amountLD = _amount;
+
+        _sendOrFallbackToHub(_oft, refundSendParam, _refundAddress);
+    }
+
+    /**
+     * @dev Send tokens via OFT with Stargate Hydra-specific error handling
+     * @dev Bridge+Swap pattern for Pool failures, retry mechanism for OFT failures
+     * @param _oft The OFT contract address to use for sending
+     * @param _sendParam The parameters for the send operation
+     * @param _refundAddress Address to receive tokens and native on Pool failure
+     */
+    function _sendOrFallbackToHub(address _oft, SendParam memory _sendParam, address _refundAddress) internal {
+        try IOFT(_oft).send{ value: msg.value }(_sendParam, MessagingFee(msg.value, 0), _refundAddress) {} catch {
+            /// @dev For Pool destinations: transfer directly to user on failure (Bridge+Swap pattern)
+            /// @dev For OFT destinations or Share tokens: revert to allow LayerZero retry mechanism
+            if (_oft == SHARE_OFT || _isOFTPath(_sendParam.dstEid)) revert OFTSendFailed(_refundAddress);
+
+            /// @dev Transfer the asset to the hub recovery address
+            IERC20(ASSET_ERC20).safeTransfer(_refundAddress, _sendParam.amountLD);
```

Comment:
We can get DoS'd here if ERC20 transfer is to an invalid address (`_refundAddress` is `address(0)`, for example).

But I don't think there's anything we want to do about it (eventually retry and hope there's credit on either source or destination).